### PR TITLE
Add threat_exception field, add update test

### PR DESCRIPTION
**Pull request by hao-nan-li**
Add threat_exception field
https://b.corp.google.com/issues/263377184

Note: This resource was not updatable previously. The current update will add the "threat_exception" field, and this is the only updatable field so far. The current update_url is hardcoded for this specific reason. If other fields become updatable, we need to figure out for other solutions.

If this PR is for Terraform, I acknowledge that I have:

- [ ] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [ ] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-the-terraform-providers), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [ ] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [ ] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/main/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [ ] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
cloudids: added `threat_exception` field to `google_cloud_ids_endpoint ` resource.

```

### CODE REVIEW

**Comment by hao-nan-li** [comment]
Looks like I got the same error as last time, need to reach out in order to do some clean up

**Comment by hao-nan-li** [comment]
b/264559773 to fix the failing test.

File: mmv1/products/cloudids/api.yaml [hunk 1]
```diff
@@ -109,3 +111,8 @@ objects:
           - :MEDIUM
           - :HIGH
           - :CRITICAL
+      - !ruby/object:Api::Type::Array
+        name: 'threat_exceptions'
```
Comment (zli82016) [inline mmv1/products/cloudids/api.yaml:115]: The name should be "threatExceptions"

File: mmv1/products/cloudids/terraform.yaml [hunk 2]
```diff
@@ -27,6 +27,8 @@ overrides: !ruby/object:Overrides::ResourceOverrides
     properties:
       location: !ruby/object:Overrides::Terraform::PropertyOverride
         ignore_read: true
+      threat_exceptions: !ruby/object:Overrides::Terraform::PropertyOverride
+        ignore_read: true
```
Comment (zli82016) [inline mmv1/products/cloudids/terraform.yaml:31]: What is the reason to set  `ignore_read: true? I think `ignore_read: true` is not needed here and can be removed.

File: mmv1/third_party/terraform/tests/resource_cloudids_endpoint_test.go [hunk 3]
```diff
@@ -25,9 +25,19 @@ func TestAccCloudIdsEndpoint_basic(t *testing.T) {
 				Config: testCloudIds_basic(context),
 			},
 			{
-				ResourceName:      "google_cloud_ids_endpoint.endpoint",
-				ImportState:       true,
-				ImportStateVerify: true,
+				ResourceName:            "google_cloud_ids_endpoint.endpoint",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"threat_exceptions"},
```
Comment (zli82016) [inline mmv1/third_party/terraform/tests/resource_cloudids_endpoint_test.go:31]: We do not need to ignore `threat_exceptions`?

File: mmv1/third_party/terraform/tests/resource_cloudids_endpoint_test.go [hunk 4]
```diff
@@ -25,9 +25,19 @@ func TestAccCloudIdsEndpoint_basic(t *testing.T) {
...
+			},
+			{
+				Config: testCloudIds_basicUpdate(context),
+			},
+			{
+				ResourceName:            "google_cloud_ids_endpoint.endpoint",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"threat_exceptions"},
```
Comment (zli82016) [inline mmv1/third_party/terraform/tests/resource_cloudids_endpoint_test.go:40]: We do not need to ignore `threat_exceptions`?

File: mmv1/third_party/terraform/tests/resource_cloudids_endpoint_test.go (hunk 3 above)
Comment (hao-nan-li) [reply mmv1/third_party/terraform/tests/resource_cloudids_endpoint_test.go:31]: I'm getting some 
    provider_test.go:307: ImportStateVerify attributes not equivalent. Difference is shown below. Top is actual, bottom is expected.
        
        (map[string]string) {
        }
        
        
        (map[string]string) (len=3) {
         (string) (len=19) "threat_exceptions.#": (string) (len=1) "2",
         (string) (len=19) "threat_exceptions.0": (string) (len=2) "12",
         (string) (len=19) "threat_exceptions.1": (string) (len=2) "67"
        }

File: mmv1/products/cloudids/api.yaml (hunk 1 above)
Comment (hao-nan-li) [reply mmv1/products/cloudids/api.yaml:115]: Nice catch, will change

File: mmv1/products/cloudids/api.yaml [hunk 5]
```diff
@@ -46,9 +46,10 @@ objects:
     name: 'Endpoint'
     base_url: 'projects/{{project}}/locations/{{location}}/endpoints'
     create_url: 'projects/{{project}}/locations/{{location}}/endpoints?endpointId={{name}}'
+    update_url: 'projects/{{project}}/locations/{{location}}/endpoints/{{name}}?update_mask=threat_exceptions'
```
Comment (zli82016) [inline mmv1/products/cloudids/api.yaml:49]: We can use  `update_mask: true` instead of hard coding update_url

https://github.com/GoogleCloudPlatform/magic-modules/blob/59946b96064bcf8db713ce4857b47e7448ccc87b/mmv1/api/resource.rb#L50

**Comment by zli82016** [comment]
Also, is the field `description` updatable? 

https://github.com/GoogleCloudPlatform/magic-modules/pull/7077/files#diff-c8637d5d3a7359ee7dd87763a5ed43275c8434c642d4cd4cf3ae47785d309fa2R89

File: mmv1/third_party/terraform/tests/resource_cloudids_endpoint_test.go (hunk 3 above)
Comment (zli82016) [reply mmv1/third_party/terraform/tests/resource_cloudids_endpoint_test.go:31]: Yeah, you can try after changing the name to `threatExceptions` in api.yam file.

File: mmv1/products/cloudids/api.yaml (hunk 5 above)
Comment (hao-nan-li) [reply mmv1/products/cloudids/api.yaml:49]: Sounds good, I'll add it.

File: mmv1/products/cloudids/terraform.yaml (hunk 2 above)
Comment (hao-nan-li) [reply mmv1/products/cloudids/terraform.yaml:31]: Yeah I think it can be removed after change the field name to threatExceptions

**Comment by hao-nan-li** [comment]
> Also, is the field `description` updatable?
> 
> https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/products/cloudids/api.yaml#L88

It should not be updatable.

**Comment by zli82016** [comment]
> > Also, is the field `description` updatable?
> > https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/products/cloudids/api.yaml#L88
> 
> It should not be updatable.

OK. Then we need to add "input: true" to the field `description` ?

**Comment by hao-nan-li** [comment]
> > > Also, is the field `description` updatable?
> > > https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/products/cloudids/api.yaml#L88
> > 
> > 
> > It should not be updatable.
> 
> OK. Then we need to add "input: true" to the field `description` ?

will do

**Review by zli82016 — APPROVED** [review]
LGTM.

**Comment by hao-nan-li** [comment]
@shuyama1 Could you help to do a sign-off? Thanks!

File: mmv1/products/cloudids/terraform.yaml [hunk 6]
```diff
@@ -28,5 +28,5 @@ overrides: !ruby/object:Overrides::ResourceOverrides
       location: !ruby/object:Overrides::Terraform::PropertyOverride
         ignore_read: true
       name: !ruby/object:Overrides::Terraform::PropertyOverride
-        custom_flatten: 'templates/terraform/custom_flatten/name_from_self_link.erb'
+        custom_flatten: 'templates/terraform/custom_flatten/id_from_name.erb'
```
Comment (shuyama1) [inline mmv1/products/cloudids/terraform.yaml:31]: question: is this change needed? what issue will it run into if we use the previous flattener?

**Review by shuyama1 — COMMENTED** [review]
LGTM in general! Just one question. 
Plus, would you mind changing `endpoint` to `google_cloud_ids_endpoint` in the changelog?

File: mmv1/products/cloudids/terraform.yaml (hunk 6 above)
Comment (hao-nan-li) [reply mmv1/products/cloudids/terraform.yaml:31]: I remember I did this change because I had some error doing an "update" for the "threat_exception" field, in the end the root cause was something else. 
Therefore I don't think this change matters much for now. 

The API team is still developing new features on this particular resource. I suggest leaving the change as it is since it  looks to work for create/update test. If this breaks in the future we can always take further action.

**Review by shuyama1 — APPROVED** [review]
